src/mlpack/bindings/python/print_wrapper_py.cpp

```diff
@@ -0,0 +1,434 @@
+  // a hyperparameter and boolean.
+  for(auto i=params.begin(); i!=params.end(); i++)
+  {
+    map<string, ParamData> methodParams = i->second.Parameters();
+
+    for(auto itr=methodParams.begin(); itr!=methodParams.end(); itr++)
+    {
+      // Checking for serializability.
+      bool isSerial;
+      i->second.functionMap[itr->second.tname]["IsSerializable"](
+          itr->second, NULL, (void*)& isSerial);
+      if(isSerial)
+        serializable.insert(itr->second.cppType);
+
+      // Checking for hyperparameter.
+      bool isHyperParam = false;
+      size_t foundArma = itr->second.cppType.find("arma");
+      if(itr->second.input && foundArma == string::npos && !isSerial)
+        isHyperParam = true;
+      hyperParams[itr->first] = isHyperParam;
+
+      // Checking for boolean.
+      if(itr->second.cppType == "bool")
+        isBool[itr->first] = true;
+      else
+        isBool[itr->first] = false;
+    }
+  }
+
+  string className = GetClassName(groupName);
+
+  // print class.
+  if(category == "regression")
+    cout << "class " << className << "(BaseEstimator)" << ":" << endl;
+  else if(category == "classification")
+  {
+    cout << "class " << className << "(BaseEstimator, ClassifierMixin)" <<
+        ":" << endl;
+  }
+  else
+    cout << "class " << className << ":" << endl;
+
+  // print the __init__ method.
+  indent += 2;
+  cout << string(indent, ' ') << "def __init__(self," << endl;
+
+  for(auto itr=hyperParams.begin(); itr!=hyperParams.end(); itr++)
+  {
+    // indent + 13, here 13 -> def __init__(
+    if(itr->second && isBool[itr->first])
+    {
+      cout << string(indent+13, ' ') << GetValidName(itr->first)
+          << " = False," << endl;
+    }
+    else if(itr->second && !isBool[itr->first])
+    {
+      cout << string(indent+13, ' ') << GetValidName(itr->first)
+          << " = None," << endl;
+    }
+  }
+
+  cout << string(indent+12, ' ') << "):" << endl;
+  cout << endl;
+
+  // storing given arguments in attributes.
+  indent += 2;
+  cout << string(indent, ' ') << "# serializable attributes." << endl;
+  if(serializable.size() == 0)
+    cout << string(indent, ' ') << "# None" << endl;
+
+  for(auto itr=serializable.begin(); itr!=serializable.end(); itr++)
+  {
+    cout << string(indent, ' ');
+    cout << "self._" << *itr << " = None" << endl;
+  }
+
+  cout << endl;
+  cout << string(indent, ' ') << "# hyper-parameters." << endl;
+
+  for(auto itr=hyperParams.begin(); itr!=hyperParams.end(); itr++)
+  {
+    if(itr->second)
+    {
+      string validName = GetValidName(itr->first);
+      cout << string(indent, ' ');
+      cout << "self." << validName << " = " << validName << endl;
+    }
+  }
+  cout << endl;
+
+  indent -= 2;
+
+  // print all method definitions.
+  for(string methodName: methods)
+  {
+    // print method name.
+    cout << string(indent, ' ') << "def " << GetMappedName(methodName);
+    cout << "(self," << endl;
```

Comment:
```suggestion
    cout << "(self, " << endl;
```
I noticed there are a lot of style issues in this file.  Do you mind taking a pass and fixing them?  Alternately, I can provide a patch commit.

Reply:
I took a pass through the file and fixed some issues. If some are still persisting feel free to point them out